docs/arch/Not Referenced Search.md

```diff
@@ -0,0 +1,15 @@
+# Not Referenced Search v1
+Adding search parameter `_not-referenced` with support for value `*:*`. [PR](https://github.com/microsoft/fhir-server/pull/4856)
+
+## Context 
+   This search parameter is needed to help users find resources that are in a bad state by not being referenced. Many users have data usage paterns that require resources to have specific connections to other resources. If a resource is missing these connections it can be hard to find and fix. This search parameter gives users a tool to help find these resources.
+
+## Decision
+   The descision was made to have the value for the search parameter be `*:*` to aline with the patern used by the FHIR standard for `_include` and `_revinclude`. While `_not-referenced` currently doestn't support other values, this gives us the option to expand the functionality in the future without a breaking change.
```

Comment:
Would be good to see an example in this section too. e.g. `Patient?_not-referenced=*:*`. 

By reading this it seems like it works on the rev-include/reverse-chained relationship only?

Can be used in the future for finding Patients who are not explicitly referenced by a specific relationship:
`/Patient?_not-referenced=Observation:subject`